### ISSUE TITLE
Base classifiers: clone should deep copy the pool of classifiers

### ISSUE THREAD

**Issue by luizgh** [Menelau/DESlib#89]
check_estimator is failing for instances, since during clone the pool of classifiers are not being properly copied. 
Need to investigate how this problem is addressed in sklearn (e.g. on sklearn.ensembles)

**Comment by luizgh** [comment]
Note: there is an issue in scikit learn that proposes a "Freeze" mechanism for estimators, that could address this issue: https://github.com/scikit-learn/scikit-learn/issues/8370

**Comment by Menelau** [comment]
@luizgh Yes, looks like freezing mechanism could address this issue (and can also be beneficial to other ensemble/meta-estimators methods). For what I check there were other ensemble libraries with the same problem.

Let's check them if that would be the best solution to our problem, if yes we will need to wait for the integration of the freezing mechanism on sklearn in order to address this issue.

**Comment by luizgh** [comment]
Agreed. For now I will just update the documentation to add a "known issue" that these estimators do not work with GridSearch  / CV methods. I will also remove this github issue from the 0.3 milestone. 

Do you have a suggestion on where to add this "Known issues" section? I was thinking that the best is to create a new section on the User guide (https://deslib.readthedocs.io/en/latest/user_guide.html). An alternative is to add it directly on the first page, but I don't think it is necessary (at least this problem will raise an Exception if people try to use a GridSearch, so the problem is not hidden from the user).

**Comment by luizgh** [comment]
Waiting on the resolution of scikit-learn/scikit-learn#8370

**Comment by Menelau** [comment]
@luizgh Do you know if there is an update on this issue?

**Comment by luizgh** [comment]
@Menelau it is on the roadmap, but from the discussions in https://github.com/scikit-learn/scikit-learn/issues/8370 it seems that it may take a lot of time. They are actually proposing a workaround for pre-trained models (see https://github.com/scikit-learn/scikit-learn/issues/13288), but I am not sure if that would address our problem
